app/common/helpers/collections.py

```diff
@@ -420,12 +433,25 @@ def get_next_question(self, current_question_id: UUID) -> Optional["Question"]:
 
         raise ValueError(f"Could not find a question with id={current_question_id} in collection={self.collection}")
 
-    def get_previous_question(self, current_question_id: UUID) -> Optional["Question"]:
+    def get_previous_question(
```

Comment:
This and `get_next_question` look almost identical, apart from the order of the iterator. Could we combine them somehow?

Reply:
Yeah I considered having something to consistently manage making the right context, I don't think they need to ever be extended (i.e a different iterator used if we made that a variable) so I think I landed on it being OK for now and something we could refactor if we end up with a more consistent way of dealing with add another contexts

Reply:
We think we'll streamline this with a decorator that manages the context overrides for helper methods that need it. Plan to do this as a refactor once we've got all the functionality/ consumers of these methods in!